full/atm_land_ice_flux_exchange.F90

```diff
@@ -1771,6 +1860,27 @@ subroutine sfc_boundary_layer ( dt, Time, Atm, Land, Ice, Land_Ice_Atmos_Boundar
             Land%tile_size, Time, mask=Land%mask)
 #endif
     endif
+
+    do tr=1,n_exch_tr
+       if (id_tr_ref(tr)>0) then
+          call fms_xgrid_get_from_xgrid (diag_atm, 'ATM', ex_tr_ref(:,tr),   xmap_sfc)  ! cjg
+          if(id_tr_ref(tr) > 0) then
+             used = fms_diag_send_data(id_tr_ref(tr),diag_atm,Time)
+          endif
+       end if
+
+       if(id_tr_ref_land(tr) > 0) then
+          call fms_xgrid_get_from_xgrid_ug (diag_land, 'LND', ex_tr_ref(:,tr), xmap_sfc)
+          !duplicate send_tile_data. We may remove id_q_ref_land in the future.
+#ifndef _USE_LEGACY_LAND_
+          call send_tile_data (id_tr_ref_land(tr), diag_land)
+#else
+          used = fms_diag_send_tile_averaged_data(id_tr_ref_land(tr), diag_land, &
+               Land%tile_size, Time, mask=Land%mask)
```

Comment:
This doesn't seem to compile:
```
/__w/FMScoupler/FMScoupler/t/null.Iu6wiF/src/coupler/full/atm_land_ice_flux_exchange.F90:1873:88:

 1873 |           call fms_xgrid_get_from_xgrid_ug (diag_land, 'LND', ex_tr_ref(:,tr), xmap_sfc)
      |                                                                                        1
Error: There is no specific subroutine for the generic 'fms_xgrid_get_from_xgrid_ug' at (1)
```

Reply:
Do you need help fixing this? I think you just need to be the call to `fms_xgrid_get_from_xgrid_ug` inside the`#ifndef _USE_LEGACY_LAND_`
https://github.com/NOAA-GFDL/FMScoupler/blob/8524c88e6aaf9204618ac38e11eced3619e51fa4/full/atm_land_ice_flux_exchange.F90#L1873-L1877